libr/util/charset.c

```diff
@@ -113,24 +113,33 @@ R_API size_t r_charset_encode_str(RCharset *rc, ut8 *out, size_t out_len, const
 
 // assumes out is as big as in_len
 R_API size_t r_charset_decode_str(RCharset *rc, ut8 *out, size_t out_len, const ut8 *in, size_t in_len) {
-	char k[32];
 	char *o = (char*)out;
-	int i;
-	for (i = 0; i < in_len; i++) {
-		ut8 ch_in = in[i];
-
-		//zero terminate the string
-		snprintf (k, sizeof (k), "%c", ch_in);//snprintf (k, sizeof (k), "0x%02x", ch_in);
-		char *v = sdb_get (rc->db_char_to_hex, k, 0);
-		memmove(v, v+2, strlen (v));
-
-		//convert to ascii
-		char str_hx[32];
-		snprintf (str_hx, sizeof (str_hx), "%c", (char) strtol( v, 0, 16));
-		const char *ret = r_str_get_fail (str_hx, "?");
-		strcpy (o, ret);
-
-		o += strlen (o);
+	size_t last_char_size;
+	size_t maxkeylen = 8;
```

Comment:
this should be rc->maxkeylen; do you want to make this change in this commit?

Reply:
It will impact r_charset_new so I will make another PR.

Reply:
Because I realize I have tons of change to make on this function.

Reply:
Sgtm, but computing the maxkeylen by iterating all the keys should be no more than 5 lines and an extra variable. But im fine if you want to make that into a separate pr. Lets finish the comments in here fitst